pkg/agent/cniserver/sriov.go

```diff
@@ -0,0 +1,153 @@
+		grpc.WithInsecure(), grpc.WithContextDialer(func(ctx context.Context, addr string) (conn net.Conn, e error) {
+			return util.DialLocalSocket(addr)
+		}))
+	if err != nil {
+		return "", fmt.Errorf("Error getting the gRPC client for pod resources: %v", err)
+	}
+
+	defer conn.Close()
+
+	client := podresourcesv1alpha1.NewPodResourcesListerClient(conn)
+	if client == nil {
+		return "", fmt.Errorf("Error getting the lister client for pod resources.")
+	}
+
+	podResources, err := client.List(ctx, &podresourcesv1alpha1.ListPodResourcesRequest{})
+	if err != nil {
+		return "", fmt.Errorf("Error getting the pod resources: %v %v", podResources, err)
+	}
+
+	var kpr KubeletPodResources
+	kpr.resources = podResources.GetPodResources()
+	podDeviceIDs := ""
+	for _, pr := range kpr.resources {
+		if pr.Name == podName && pr.Namespace == podNamespace {
+			for _, ctr := range pr.Containers {
+				for _, dev := range ctr.Devices {
+					podDeviceIDs += fmt.Sprint(dev.DeviceIds)
+				}
+				if len(podDeviceIDs) > 2 {
+					// example: [0000:81:02.0 0000:81:0a.3]
+					// remove "[]" and return DeviceIds
+					length := len(podDeviceIDs) - 1
+					podDeviceIDs = podDeviceIDs[1:length]
+					klog.V(2).Info("pod DeviceIds of ", podName, podNamespace, "are: ", podDeviceIDs)
+					return podDeviceIDs, nil
+				}
+			}
+		}
+	}
+
+	return "", nil
+}
+
+func GetPodPCIDeviceIDRequest(args *skel.CmdArgs) string {
+	// Parse pod resources list and get the PCI device for K8S_POD_NAME and K8S_POD_NAMESPACE
+	os.Setenv("POD_DEVICE_ID", "")
+
+	k8sArgs := &K8sArgs{}
+	err := types.LoadArgs(args.Args, k8sArgs)
+	if k8sArgs != nil && err == nil {
+		deviceID, err := GetPodResourceRequests(string(k8sArgs.K8S_POD_NAME), string(k8sArgs.K8S_POD_NAMESPACE))
+		if deviceID != "" && err == nil {
+			os.Setenv("POD_DEVICE_ID", deviceID)
+		}
+	}
+	return os.Getenv("POD_DEVICE_ID")
+}
+
+// GetVFInfo takes in a VF's PCI device id and returns its PF and VF ID.
+func GetVFInfo(vfPCI string) (string, int, error) {
+	var vfID int
+
+	pf, err := sriovcniutils.GetPfName(vfPCI)
+	if err != nil {
+		return "", vfID, err
+	}
+
+	vfID, err = sriovcniutils.GetVfid(vfPCI, pf)
+	if err != nil {
+		return "", vfID, err
+	}
+
+	return pf, vfID, nil
+}
+
+// GetVFLinkNames returns VF's network interface name given it's PCI address.
+func GetVFLinkNames(pciAddress string) (string, error) {
+	return sriovcniutils.GetVFLinkNames(pciAddress)
+}
+
+// configureSecondaryInterface adds Secondary Interface support.
+// Limitation: only SR-IOV interface is supported as of now.
+func (pc *podConfigurator) configureSecondaryInterface(
+	podName string,
+	podNameSpace string,
+	containerID string,
+	containerNetNS string,
+	containerIFDev string,
+	mtu int,
+	podSriovVFDeviceID string,
+	result *current.Result,
+) error {
+	err := pc.ifConfigurator.configureContainerLink(podName, podNameSpace, containerID, containerNetNS, containerIFDev, mtu, "", podSriovVFDeviceID, result)
```

Comment:
Probably we should return an error when podSriovVFDeviceID is "", as we support only SRIOV now.

Reply:
Done